core/embed/extmod/modtrezorcrypto/modtrezorcrypto-secp256k1_zkp.h

```diff
@@ -325,13 +331,265 @@ STATIC mp_obj_t mod_trezorcrypto_secp256k1_context_multiply(
                       secp256k1_ecdh_hash_passthrough, NULL)) {
     mp_raise_ValueError("Multiply failed");
   }
+  // Verify and re-serialize into (optionally compressed) public key.
+  size_t outlen = sizeof(out);
+  if (!secp256k1_ec_pubkey_parse(ctx, &ec_pk, out, outlen)) {
```

Comment:
"shared public key" sounds like this a public then. Maybe call it "shared secret group element"  instead.

Reply:
After removing the parsing at https://github.com/romanz/trezor-firmware/pull/1/commits/729918821371de7462160ddc5e18f55944e26871, the error message is not needed.